python/ray/util/client/server/server.py

```diff
@@ -735,6 +735,20 @@ def ray_connect_handler(job_config: JobConfig = None, **ray_init_kwargs):
     return ray_connect_handler
 
 
+def try_create_gcs_client(address: Optional[str]) -> Optional[GcsClient]:
+    """
+    Try to create a gcs client based on the the command line args or by
+    autodetecting a running Ray cluster.
+    """
+    if address is None:
```

Comment:
This may not work when there are multiple running Ray clusters. Why is address not always specified?

Reply:
I think if there are multiple running ray clusters it has to be specified, just like `ray status` won't work when there are two ray clusters running.

server can be started by `python -m ray.util.client.server` and the default value is None and it'll infer the actual value. Check the try_create_redis_client in the same file.

Reply:
Do you know which specific test is calling `python -m ray.util.client.server` directly without an address? An error should be logged in this case. The Redis logic might be there for legacy reasons. For the common case of Ray launching client server, `address` should always be specified.

Reply:
I'm not sure about this is the right direction to go. Any reason to go this way instead of keeping the current redis ray?

Reply:
Since we are looking at this code path, I'm just surprised that `address` can be `None` and I'm not sure if it is the behavior we want to support long term. Which test is this fixing, `test_client_reconnect`?

Reply:
A comment explaining that `address =None` is unexpected by tested somewhere would work too.

Reply:
I think either way it should work technically. The test fixing is health check.
I'm just not sure it's the right direction to go. Also until cleaning up, you can't make address a required value.

Reply:
I see that `test_healthcheck_ray_client_server` indeed starts its own client server. The right fix should be to call `canonicalize_bootstrap_address()` on the `address` flag value, which is how `ray.init()`, `ray start` and other commands handle `address`.